keras_cv/src/models/object_detection/mask_rcnn/mask_rcnn.py

```diff
@@ -0,0 +1,515 @@
+from keras_cv.src.models.object_detection.mask_rcnn.faster_rcnn_backbone import (
+    BOX_VARIANCE,
+)
+from keras_cv.src.models.object_detection.mask_rcnn.faster_rcnn_backbone import (
+    _parse_box_loss,
+)
+from keras_cv.src.models.object_detection.mask_rcnn.faster_rcnn_backbone import (
+    _parse_classification_loss,
+)
+from keras_cv.src.models.object_detection.mask_rcnn.faster_rcnn_backbone import (
+    _parse_rpn_classification_loss,
+)
+from keras_cv.src.models.object_detection.mask_rcnn.faster_rcnn_backbone import (
+    unpack_input,
+)
+from keras_cv.src.models.object_detection.mask_rcnn.mask_head import MaskHead
+from keras_cv.src.models.object_detection.mask_rcnn.roi_sampler import (
+    ROISampler,
+)
+from keras_cv.src.models.task import Task
+
+
+@keras_cv_export(
+    [
+        "keras_cv.models.MaskRCNN",
+        "keras_cv.models.object_detection.MaskRCNN",
+    ]
+)
+class MaskRCNN(Task):
+    """A Keras model implementing the Mask R-CNN architecture.
+
+    Mask R-CNN is a straightforward extension of Faster R-CNN, providing an
+    additional mask head that predicts segmentation masks. The constructor
+    therefore requires a FasterRCNN model as backend.
+
+    This model is compatible with Keras 3 only.
+
+    Args:
+        backbone: `keras.Model`. A FasterRCNN backbone model.
+        mask_head: (Optional) A `keras.Layer` that performs regression of
+            the segmentation masks. If not provided, a network with
+            2 convolutional layers, an upsampling layer and a class-specific
+            layer will be used.
+    """  # noqa: E501
+
+    def __init__(self, backbone, mask_head=None, **kwargs):
+        mask_head = mask_head or MaskHead(
+            backbone.num_classes, name="mask_head"
+        )
+
+        images = backbone.input
+        feature_map = backbone.get_layer("roi_align").output
+
+        segmask_pred = mask_head(feature_map=feature_map)
+        segmask_pred = keras.layers.Concatenate(axis=1, name="segmask")(
+            [segmask_pred]
+        )
+
+        inputs = {"images": images}
+        outputs = {"segmask": segmask_pred, **backbone.output}
+
+        super().__init__(
+            inputs=inputs,
+            outputs=outputs,
+            **kwargs,
+        )
+
+        self.backbone = backbone
+        self.mask_head = mask_head
+        self.roi_sampler = ROISampler(
+            roi_bounding_box_format="yxyx",
+            gt_bounding_box_format=backbone.bounding_box_format,
+            roi_matcher=backbone.box_matcher,
+            num_sampled_rois=backbone.roi_sampler.num_sampled_rois,
+            mask_shape=(14, 14),
+        )
+
+    def compile(
+        self,
+        rpn_box_loss=None,
+        rpn_classification_loss=None,
+        box_loss=None,
+        classification_loss=None,
+        mask_loss=None,
+        weight_decay=0.0001,
+        loss=None,
+        metrics=None,
+        **kwargs,
+    ):
+        if loss is not None:
```

Comment:
add comments to mention section of code computing loss

Reply:
Done.